### PR TITLE
2454 indicate expired pano

### LINKED ISSUE #2454: How many pano ids are expired in Newberg and ways of making that clear in the LabelMap tool?
Do we have a sense of how many pano ids have expired on Newberg? I was trying to show [LabelMap](https://sidewalk-newberg.cs.washington.edu/labelmap) but ended up _not_ showing it because when I clicked on like ~10 random points, about 7 showed me the following message (so I got the sense that most of Newberg has expired panos now):

image

Later, after my phone call, I went through and investigated a bit more and now I'm not getting that message as much. So maybe just bad luck?

Perhaps we could outline the label circles with expired panos a different color to get a sense of the problem and to easily avoid clicking on those in the future?

### PR DESCRIPTION
**Pull request by mcdoerr**
Resolves #2454 

Adding a new look for a pano that has expired imagery. We indicate this by moving the color to be a .5 opacity border, with a light grey center. 

To implement this, changing the Slick to include an additional join with the gsv_data table to include the expired table. Locally, this doesn't seem to affect performance on a noticeable level. The type is BooleanOption because some columns may be null (according to errors that arose from just using Boolean). 

I spot tested that the expired labels match the labels that don't have imagery.

Here is an image of the expired panos, with some non-expired panos for contrast.
Screen Shot 2021-03-08 at 12 36 33 PM

### CODE REVIEW

File: app/models/label/LabelTable.scala [hunk 1]
```diff
@@ -1170,19 +1171,18 @@ object LabelTable {
     */
   def selectLocationsAndSeveritiesOfLabels: List[LabelLocationWithSeverity] = db.withSession { implicit session =>
     val _labels = for {
-      (_labels, _labelTypes) <- labelsWithoutDeleted.innerJoin(labelTypes).on(_.labelTypeId === _.labelTypeId)
-    } yield (_labels.labelId, _labels.auditTaskId, _labels.gsvPanoramaId, _labelTypes.labelType, _labels.panoramaLat, _labels.panoramaLng)
```
Comment (misaugstad) [inline app/models/label/LabelTable.scala:1174]: Thanks for noticing that we weren't even using `panoramaLat` and `panoramaLng`!  This query was written 4 years ago, and from the beginning they were never used :upside_down_face:

File: public/javascripts/Choropleths/InitializeSubmittedLabels.js [hunk 2]
```diff
@@ -105,7 +105,10 @@ function InitializeSubmittedLabels(map, params, adminGSVLabelView, mapData, labe
             pointToLayer: function (feature, latlng) {
                 let style = $.extend(true, {}, geojsonMarkerOptions);
                 style.fillColor = colorMapping[feature.properties.label_type].fillStyle;
-                if (params.includeLabelColor) {
+                if (feature.properties.expired) {
```
Comment (misaugstad) [inline public/javascripts/Choropleths/InitializeSubmittedLabels.js:108]: What happens if `params.includeLabelColor` is `false`, but `feature.properties.expired` is `true`? Then I think you'd be unintentially changing the look of those maps. You should probably just do:
```
if (params.includeLabelColor) {
    if (feature.properties.expired) {
        style.fillColor = 'lightgrey';
        style.color = colorMapping[feature.properties.label_type].missingPanoStrokeStyle;
    } else {
        style.color = colorMapping[feature.properties.label_type].strokeStyle;
    }
}
```

**Review by misaugstad — CHANGES_REQUESTED** [review]
Okay, so you noticed a problem in the query here (some columns were being included and then never used). And it made me realize that this query was looking pretty bad and I couldn't help myself and I did a quick rewrite of it :sweat_smile: It's a little shorter and simpler. And instead of setting the latitude/longitude to 0 if they are null, it just filters them out now. Setting them to 0 just doesn't make sense :shrug: 

>The type is BooleanOption because some columns may be null (according to errors that arose from just using Boolean).

I switched the join you made from a "left join" to an "inner join", which means that a row is included in the join only if there is an entry in _both_ tables that matches the description (`_l.gsvPanoramaId === _gsv.gsvPanoramaId`). So now the `expired` column doesn't need to be an Option type! I made that change during my rewrite. Nice work on figuring out how to do all of that though!

Okay, now on to the visual stuff! The issue I'm having is that the labels with expired imagery seem to "pop" more when the map is zoomed out (which is how the page loads). Adding some example images below. Maybe we need to try decreasing the opacity of the border a bit more? Or decreasing it's thickness? Not sure. But I think that future iterations should have both zoomed in and zoomed out screenshots.

Screenshot from 2021-03-10 14-41-36
Screenshot from 2021-03-10 14-40-26

**Comment by mcdoerr** [comment]
Yeah, I agree about the panos popping, but I think what's actually popping is the "Other" category, which has a gray (instead of light gray) center and the border is not transparent at all. Here's a non-zoomed image without the expired pano new look, so what is currently on the website of the "Other" pano popping. 
Screen Shot 2021-03-10 at 4 06 56 PM

**Comment by misaugstad** [comment]
Oops, thank you! I think that I had a No Sidewalk and an Other label next to each other, and accidentally clicked on the No Sidewalk label twice, so I thought that that's what I was looking at, my bad!! Let me take another look real quick.

**Comment by mcdoerr** [comment]
It seems to me that the Other category when stacked on top of each other because it's very zoomed creates a lot of contrast since the opacities stack to produce a very dark grey. Not sure what the fix for that would look like, seems like maybe another issue?

I'm a tad worried about expired panos being confused with "Other" or "No Sidewalk" though

**Comment by misaugstad** [comment]
I'm playing around with the opacity, and now that I'm seeing it zoomed out as well, I think that I like something more like 0.8. Here are a couple pics. The difference between "Other" and the labels with expired imagery is clear here as well.

Screenshot from 2021-03-10 16-30-24
Screenshot from 2021-03-10 16-31-36
Screenshot from 2021-03-10 16-31-51
Screenshot from 2021-03-10 16-32-08

**Comment by misaugstad** [comment]
@mcdoerr how do those look to you? you had said that you prefer the opacity of 0.5, do you still prefer that?

**Comment by mcdoerr** [comment]
I was looking at it some more, and I've found some labels that have imagery but have the expired column marked true for some reason (verified it is true in the database). Any idea why that might be happening? 

One of the panos I found with this issue is 'OEjCJD6XfUnADm8KaIoIvw'.

I still like the .5 because I think it makes it less likely to be clicked on, but I don't feel super strongly about it!

**Comment by misaugstad** [comment]
>I still like the .5 because I think it makes it less likely to be clicked on, but I don't feel super strongly about it!

I think that .5 can sometimes just make it look like there is no color at all. It's kind of hard to see any purple here:
Screenshot from 2021-03-10 16-47-00

**Comment by misaugstad** [comment]
>I was looking at it some more, and I've found some labels that have imagery but have the expired column marked true for some reason (verified it is true in the database). Any idea why that might be happening?

Hmm... It's possible that when we checked if there was imagery at some point, there may have been some network error or something. And it might be hard to distinguish between that and Google just not having the imagery available. So I suppose it's possible that a small number of images are marked incorrectly that way. I think that when we implement #2458 we should consider also looking at images that have been marked as expired to double check!

**Comment by mcdoerr** [comment]
Oh yeah that screenshot definitely convinces me, I'll update it to .8, I agree

**Comment by mcdoerr** [comment]
Ok, that sounds good regarding #2458!

**Review by misaugstad — APPROVED** [review]
Looks good!! Thanks @mcdoerr !

**Pull request merged**